packages/emitter-framework/test/typescript/components/union-declaration.test.tsx

```diff
@@ -1,182 +1,269 @@
             </SourceFile>
           </Output>,
         );
 
-        const testFile = res.contents.find((file) => file.path === "test.ts");
-        assert(testFile, "test.ts file not rendered");
-        const actualContent = await format(testFile.contents as string, { parser: "typescript" });
-        const expectedContent = await format(`let x:"one" | "two" = "one"`, {
-          parser: "typescript",
+        assertFileContents(
+          res,
+          d`
+            let x: "one" | "two" = "one";
+          `,
+        );
+      });
+
+      describe("Discriminated Union", () => {
+        it("renders a discriminated union declaration", async () => {
+          const { TestUnion } = (await runner.compile(`
+          namespace DemoService;
+          @discriminated
+          @test union TestUnion {
+            one: { oneItem: true },
+            two: true
+          }
+        `)) as { TestUnion: Union };
+
+          const res = render(
+            <Output program={runner.program}>
+              <SourceFile path="test.ts">
+                <UnionDeclaration type={TestUnion} />
+              </SourceFile>
+            </Output>,
+          );
+
+          assertFileContents(
+            res,
+            d`
+            type TestUnion = {
+              kind: "one";
+              value: {
+                oneItem: true;
+              };
+            } | {
+              kind: "two";
+              value: true;
+            };
+          `,
+          );
         });
-        expect(actualContent).toBe(expectedContent);
       });
-    });
 
-    describe("Bound to Enum", () => {
-      it("creates a union declaration", async () => {
-        const program = await getProgram(`
+      it("renders a discriminated union declaration with custom properties", async () => {
+        const { TestUnion } = (await runner.compile(`
         namespace DemoService;
-        enum TestEnum {
-          one: "one",
-          two: "two"
+        @discriminated(#{ discriminatorPropertyName: "dataKind", envelopePropertyName: "data" })
+        @test union TestUnion {
+          one: { oneItem: true },
+          two: true
         }
-        `);
+      `)) as { TestUnion: Union };
+
+        const res = render(
+          <Output program={runner.program}>
+            <SourceFile path="test.ts">
+              <UnionDeclaration type={TestUnion} />
+            </SourceFile>
+          </Output>,
+        );
+        assertFileContents(
+          res,
+          d`
+          type TestUnion = {
+            dataKind: "one";
+            data: {
+              oneItem: true;
+            };
+          } | {
+            dataKind: "two";
+            data: true;
+          };
+        `,
+        );
+      });
 
-        const [namespace] = program.resolveTypeReference("DemoService");
-        const union = Array.from((namespace as Namespace).enums.values())[0];
+      it("renders a discriminated union declaration with no envelope", async () => {
```

Comment:
what does it looks like in thoses cases when you use a named model?

Reply:
Using the example from the typespec docs:

```typespec
@discriminated(#{ envelope: "none" })
union Pet {
  cat: Cat,
  dog: Dog,
}

model Cat {
  name: string;
  meow: boolean;
}

model Dog {
  name: string;
  bark: boolean;
}
```

Renders as:

```typescript
type TestUnion = {
  kind: "cat";
  name: string;
  meow: number;
} | {
  kind: "dog";
  name: string;
  bark: string;
};
```